server/routes/api.js

```diff
@@ -23,13 +24,55 @@ module.exports = (services) => {
       throw err;
     }
   });
-
+  /* POST calculate */
   router.post('/calculate', (req, res) => {
     const dijkstra = new Dijkstra(req.body.from, req.body.dest, { MAX_DISTANCE: 4 });
     const result = dijkstra.solve();
     console.log(result);
     res.json(result);
   });
+  /* POST routes */
+  router.post('/routes', async (req, res) => {
+    try {
+      const routes = req.body;
```

Comment:
req.body will be an object, so you need to write `const routes = req.body.routes;`

Reply:
@bill-feng Fixed it!